app/validators/email_validator.rb

```diff
@@ -0,0 +1,9 @@
+# app/validators/email_validator.rb
+
+class EmailValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    unless value =~ /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i

```

Comment:
I am not sure about this regex. To me it looks like some wild characters are allowed and in general we do not focus on this topic because the right regex is just couple lines long statement. In Railstutorial from Michael Hartl is a nice way how to Regex this.